### PR TITLE
Fixed bug with validation threshold generated for mean diff

### PR DESCRIPTION
**Pull request by JRosenkranz**
This PR fixes a bug where mean diff metric was being calculated incorrectly resulting in smaller diffs than expected.

### CODE REVIEW

**Comment by JRosenkranz** [comment]
bot:test
TEST_FILE=test_decoders.py MODEL_ID=ibm-granite/granite-3.2-8b-instruct BATCH_SIZE=1 SEQUENCE_LENGTH=64 USE_TINY_MODEL=0

**Comment by JRosenkranz** [comment]
bot:test
TEST_FILE=test_decoders.py MODEL_ID=ibm-granite/granite-3.2-8b-instruct BATCH_SIZE=1 SEQUENCE_LENGTH=64 USE_TINY_MODEL=0

File: scripts/generate_metrics.py [hunk 1]
```diff
@@ -108,7 +127,7 @@
         extra_get_model_kwargs[a_split[0]] = a_split[1]
 
 # this follows the same pattern of naming in test_shapes. This way we can save and re-use for quicker shape testing.
-prefix = f"{args.variant.replace('/', '--')}_max-new-tokens-{args.max_new_tokens}_batch-size-{args.batch_size}_seq-length-{args.min_pad_length}_dtype-{args.default_dtype}"
+prefix = get_default_validation_prefix(args.variant, args.max_new_tokens, args.batch_size, args.min_pad_length, args.default_dtype)
```
Comment (ani300) [inline scripts/generate_metrics.py:130]: should we add world_size to this prefix?

File: tests/models/test_decoders.py [hunk 2]
```diff
@@ -251,9 +251,9 @@ def __filter_before_eos(l, filter_indexes):
 
 
 def __get_validation_info_full_path(
-    model_path, batch_size, seq_length, max_new_tokens, seed
+    model_path, batch_size, seq_length, max_new_tokens, seed, device_type="cpu"
 ):
-    validation_file_name = f"{model_path.replace('/', '--')}_max-new-tokens-{max_new_tokens}_batch-size-{batch_size}_seq-length-{seq_length}_dtype-fp16.cpu_validation_info.{seed}.out"
+    validation_file_name = f"{get_default_validation_prefix(model_path, max_new_tokens, batch_size, seq_length, 'fp16')}.{device_type}_validation_info.{seed}.out"
```
Comment (ani300) [inline tests/models/test_decoders.py:256]: should this have the world_size if we're doing distributed tests too?

File: aiu_fms_testing_utils/testing/validation.py [hunk 3]
```diff
@@ -88,6 +88,9 @@ def save(self, save_dir_path: str):
 
     def __len__(self):
         return len(self._validation_info_list)
+    
+def get_default_validation_prefix(model_id: str, max_new_tokens: int, batch_size: int, seq_length: int, dtype: str):
```
Comment (ani300) [inline aiu_fms_testing_utils/testing/validation.py:92]: maybe add the world size here too

**Review by ani300 — APPROVED** [review]
lgtm, small questions about distributed tests

File: scripts/generate_metrics.py (hunk 1 above)
Comment (JRosenkranz) [reply scripts/generate_metrics.py:130]: The outputs should not change while using tp. This is mostly for the cpu logits
